flask/boxwise_flask/models/box.py

```diff
@@ -110,38 +110,20 @@ def __unicode__(self):
         return self.box_label_identifier
 
     @staticmethod
-    def create_box(box_creation_input):
+    def get_box(box_id):
+        return Box.get(Box.box_label_identifier == box_id)
 
-        today = datetime.now()
-        barcode = box_creation_input.get("qr_barcode", None)
-        qr_id_from_table = QRCode.get_id_from_code(barcode)
-        box_uuid = uuid.uuid4()
-        box_short_uuid = str(box_uuid)[
-            :11
-        ]  # the table is truncating a full uuid to 11 chars, so do it preemptively
 
-        new_box = Box.create(
-            # surprisingly not primary key, unique non-sequential identifier for a box
-            box_label_identifier=box_short_uuid,
-            product_id=box_creation_input.get(
-                "product_id", None
-            ),  # will become a fancy dropdown on the FE
-            size_id=box_creation_input.get(
-                "size_id", None
-            ),  # will be tied to the product_id lookup somehow
-            items=box_creation_input.get("items", None),
-            location_id=box_creation_input.get(
-                "location_id", None
-            ),  # based on the user's allowed bases
-            comment=box_creation_input.get("comment", None),
-            qr_id=qr_id_from_table,
-            created=today,
-            # this is consistently NULL in the table, do we want to change that?
-            created_by=box_creation_input.get("created_by", None),
-            box_state_id=1,  # always 1 for create?
-        )
-        return new_box
+def create_box(box_creation_input):
+    today = datetime.now()
+    qr_code = box_creation_input.pop("qr_code", None)
+    qr_id = QRCode.get_id_from_code(qr_code)
```

Comment:
@aerinsol is it true that a QR code is always generated _before_ a box is created? Can a box be created _without_ a QR code at all?

Reply:
@pylipp As far as I know, the QR code generated separately and printed as label will later be scanned and assigned to the box, so I think this should be considered as required input for the create_box method. Hope this helps.

Reply:
Yes that's correct, currently it's possible to generate a box without having a qrcode attached to it. We are thinking of discontinuing this feature, but since this behavior is currently allowed we should not force this validation on the back-end.